engines/mock/mockengine.go

```diff
@@ -7,19 +7,22 @@ import (
 
 	"github.com/taskcluster/taskcluster-worker/engines"
 	"github.com/taskcluster/taskcluster-worker/engines/extpoints"
+	log "github.com/taskcluster/taskcluster-worker/log"
 	"github.com/taskcluster/taskcluster-worker/runtime"
 )
 
 type engine struct {
 	engines.EngineBase
+	Log log.Logger
 }
 
 func init() {
 	// Register the mock engine as an import side-effect
 	extpoints.EngineProviders.Register(func(
-		options extpoints.EngineOptions,
+		options *extpoints.EngineOptions,
 	) (engines.Engine, error) {
-		return engine{}, nil
+		logger := options.Environment.Log.WithField("engine", "mock")
```

Comment:
> I for the record: Log wasn't added to EngineOptions :)

Forget that comment I was wrong... it's part of the runtime environment... Hmm... it probably should definitely be there...
But how about we add a child logger to `EngineOptions` when we create it... It'll be a child logger of the logger on the runtime environment.